From f4c10758f2e53cd89d6105995757ba3606acdeda Mon Sep 17 00:00:00 2001
From: Hendrik Brueckner <brueckner@redhat.com>
Date: Wed, 23 Jul 2014 10:50:04 -0400
Subject: [virt] virtio_net: don't crash if virtqueue is broken

Message-id: <1406112611-9862-17-git-send-email-brueckner@redhat.com>
Patchwork-id: 86327
O-Subject: [RHEL7.1 PATCH 16/23] [virt] virtio_net: don't crash if virtqueue is broken.
Bugzilla: 1088554
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

Description
-----------
commit a7c58146cf9a782113629021ba5420582fef265e
Author: Rusty Russell <rusty@rustcorp.com.au>
Commit: Rusty Russell <rusty@rustcorp.com.au>

    virtio_net: don't crash if virtqueue is broken.

    A bad implementation of virtio might cause us to mark the virtqueue
    broken: we'll dev_err() in that case, and the device is useless, but
    let's not BUG_ON().

    Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>

Bugzilla
--------
BZ 1088554
https://bugzilla.redhat.com/show_bug.cgi?id=1088554

Upstream status of the patch
----------------------------
The patch is upstream as of kernel version v3.15-rc1
http://git.kernel.org/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commitdiff;h=a7c58146cf9a782113629021ba5420582fef265e

Brew
----
https://brewweb.devel.redhat.com/taskinfo?taskID=7725759

Test status
-----------
The patch has been tested and fixes the problem.
The fix has been verified by the IBM test department.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index ddc1d2b..72d3c0f 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -858,8 +858,7 @@ static bool virtnet_send_command(struct virtnet_info *vi, u8 class, u8 cmd,
  sgs[out_num + in_num++] = &stat;
 
  BUG_ON(out_num + in_num > ARRAY_SIZE(sgs));
- BUG_ON(virtqueue_add_sgs(vi->cvq, sgs, out_num, in_num, vi, GFP_ATOMIC)
-        < 0);
+ virtqueue_add_sgs(vi->cvq, sgs, out_num, in_num, vi, GFP_ATOMIC);
 
  virtqueue_kick(vi->cvq);
 
-- 
1.7.1